aqt/archives.py

```diff
@@ -407,12 +406,13 @@ def _parse_update_xml(self, os_target_folder, update_xml_text, target_packages:
         for packageupdate in package_updates:
             if not self.all_extra:
                 target_packages.remove_module_for_package(packageupdate.name)
-            should_filter_archives: bool = self.subarchives and self.should_filter_archives(packageupdate.name)
+            should_filter_archives: bool = (self.subarchives is not None) and self.should_filter_archives(packageupdate.name)
```

Comment:
This changes condition when `self.subarchives` is empty list

```suggestion
            should_filter_archives: bool = bool(self.subarchives) and self.should_filter_archives(packageupdate.name)
```